core/src/com/unciv/logic/city/CityStats.kt

```diff
@@ -309,6 +311,20 @@ class CityStats {
                 stats.production += 15
         }
 
+        for (unique in cityInfo.cityConstructions.builtBuildingUniqueMap.getUniques("+[]% production when building [] in this city")) {
+            val filter = unique.params[1]
+            if (currentConstruction.name == filter
+                    || (filter == "land units" && currentConstruction is BaseUnit && currentConstruction.unitType.isLandUnit())
+                    || (filter == "naval units" && currentConstruction is BaseUnit && currentConstruction.unitType.isWaterUnit())
+                    || (filter == "mounted units" && currentConstruction is BaseUnit && currentConstruction.unitType == UnitType.Mounted)
```

Comment:
Same with this - unless we plan on making this a separate unique that accepts a unitType
"+[]% production when building [] units in this city"
This connects to another big dilemma I have about unit filtering as a whole